src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/BookingService.kt

```diff
@@ -165,13 +166,20 @@ class BookingService(
       // Bookings will need to be specialised in a similar way to Premises so that TA Bookings do not have a keyWorkerStaffCode field
       check(premises is ApprovedPremisesEntity) { "Booking has a Key Worker specified but Premises is not an ApprovedPremises" }
 
-      val staffMemberResult = staffMemberService.getStaffMemberByCode(keyWorkerStaffCode, premises.qCode)
-
-      if (staffMemberResult !is AuthorisableActionResult.Success) {
-        throw InternalServerErrorProblem("Unable to get Key Worker via Staff Code: $keyWorkerStaffCode / Q Code: ${premises.qCode}")
+      when (val staffMemberResult = staffMemberService.getStaffMemberByCode(keyWorkerStaffCode, premises.qCode)) {
```

Comment:
i think we may need to do a similar thing for PremisesController#premisesPremisesIdBookingsGet? although i'm not sure if that endpoint is used?

Reply:
Nah, we don't use that anymore, as we have the premisesSummary stuff to return a cleaner summary of bookings and premises info